tests/security/pam/pam_faillock.pm

```diff
@@ -0,0 +1,97 @@
+# Copyright 2020 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Summary: PAM tests for faillock, the uesr login can be locked
+#          if reaches the limit number of authentication failures;
+#          we can unlock it as well with root user.
+# Maintainer: rfan1 <richard.fan@suse.com>
+# Tags: poo#102990 tc#1769824
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use base 'consoletest';
+use utils 'zypper_call';
+use version_utils 'is_sle';
+
+sub run {
+    select_console('root-console');
+    zypper_call('in expect');
+
+    my $user_name = 'pamtest';
+    my $user_pw = $testapi::password;
+    my $bad_pw = 'badpassword';
+    assert_script_run("useradd -m $user_name");
+    assert_script_run("echo $user_name:$user_pw | chpasswd");
+
+    # Package change log check on SLE: jsc#sle-20638
+    validate_script_output('rpm -q pam --changelog', sub { m/jsc#sle-20638/ }) if (is_sle);
+
+    # Basic function test, lock&unlock
+    # Modify the pam configuration files
+    my $pam_config = <<EOF;
+auth       required        pam_faillock.so preauth deny=3 unlock_time=600
+auth       required        pam_faillock.so authfail deny=3 unlock_time=600
+account    required        pam_faillock.so
+EOF
+    assert_script_run("cp /etc/pam.d/common-auth /etc/pam.d/common-auth.back");
+    assert_script_run("cp /etc/pam.d/common-password /etc/pam.d/common-password.back");
+    assert_script_run("echo '$pam_config' >> /etc/pam.d/common-auth");
+    assert_script_run("echo '$pam_config' >> /etc/pam.d/common-password");
```

Comment:
I recommend you use variables to define files, such as  /etc/pam.d/common-auth, /etc/pam.d/common-password

Reply:
I have the opposite option here, we don't need to set a variable for system configuration file, we used more lines and get less benefit, and make code not friendly to be read. :)

If it is not a must one, I will keep current logic.

Reply:
Sure, I understood what you mean, but defining a variable is really a good way, such as if file name changed in the future you only need to revise the definition line to update. Besides if the variable's name is well named it is also readable too :).
Anyway, it is optional you can “keep current logic". ：）